src/intervaloptimisation.jl

```diff
@@ -3,58 +3,6 @@ Methods using Interval Optimisation
 ===================================#
 using .IntervalOptimisation
 
-"""
-    range_and_optimisers(f::Function, dom::Interval_or_IntervalBox;
-                         structure=HeapedVector, tol=1e-3)
-
-Return an interval that contains the global minimum (resp. an interval that
-contains the global maximum) of a univariate or multivariate function `f`,
-as well as a list of boxes containing the minimisers (resp. minimisers).
-
-### Input
-
-- `f`         -- function
-- `dom`       -- hyperrectangular domain, either a unidimensional  `Interval` or
-                 a multidimensional `IntervalBox`
-- `structure` -- (optional, default: `HeapedVector`) the way in which vector elements
-                 are kept arranged; possible options are `HeapedVector` and `SortedVector`
-- `tol`       -- tolerance to which the optima are computed
-
-### Output
-
-The 4-tuple `(global_min, minimisers, global_max, maximisers)`, where:
-
-- `global_min`: interval that is guaranteed to contain the global minimum of `f`
-                over `dom`
-- `minimisers`: list of boxes containing the minimisers of `f` over `dom`
-- `global_max`: interval that is guaranteed to contain the global maximum of `f`
-                over `dom`
-- `maximisers`: list of boxes containing the maximisers of `f` over `dom`
-
-### Algorithm
-
-The solver finds the global minimum of the function `f` over the `Interval` or
-`IntervalBox` `dom` using the Moore-Skelboe algorithm.
-
-We refer to the documentation and source code of `IntervalOptimisation` for details.
-"""
-function range_and_optimisers(f::Function, dom::Interval_or_IntervalBox;
-                              structure=HeapedVector, tol=1e-3)
-    _f = wrap_f(f, dom)
-    return _range_and_optimisers(_f, dom, structure, tol)
-end
-
-@inline wrap_f(f::Function, dom::Interval) = f
-
-# for multidimensional domains, it is required that f has a single argument
-@inline wrap_f(f::Function, dom::IntervalBox) = X -> f(X...)
```

Comment:
The main difference is that now `X -> f(X...)` is also used for `Interval`s. Does that still work?

Reply:
the tests are still passing at least :), the same pattern was already used e.g. [here](https://github.com/JuliaReach/RangeEnclosures.jl/blob/master/src/intervals.jl#L29-L31)

Reply:
```julia
julia> g(X...) = for x in X println(x) end
g (generic function with 1 method)

julia> f(X) = g(X...)
f (generic function with 1 method)

julia> f(1..2)
[1, 2]

julia> f(IntervalBox(1..2, 3..4, 5..6))
[1, 2]
[3, 4]
[5, 6]
```

basically `IntervalBox` has the array interface defined for it so `X...` will destruct it into distinct arguments, `Interval` not, so `X...` does nothing to it